pandas/_libs/groupby.pyx

```diff
@@ -1350,6 +1350,10 @@ cdef group_min_max(iu_64_floating_t[:, ::1] out,
                     else:
                         if uses_mask:
                             result_mask[i, j] = True
+                            # set out[i, j] to 0 to be deterministic, as
```

Comment:
shouldn't you just np.zeros instead of empyt to start?

Reply:
empty is cheaper in the non-masked case